pkg/controller/virtualnode.go

```diff
@@ -512,6 +518,41 @@ func (c *Controller) mutateVirtualNodeForProcessing(vnode *appmeshv1beta1.Virtua
 			}
 		}
 	}
+
+	// The following overrides facilitate matching the spec to the API response.
+	// They can be removed if the API response matches the input spec.
+	if vnode.Spec.Backends != nil {
+		for _, backend := range vnode.Spec.Backends {
+			if backend.VirtualService.ClientPolicy != nil {
+				clientPolicy := backend.VirtualService.ClientPolicy
+				if clientPolicy.Tls != nil {
+					mergeTlsClientPolicyDefaults(clientPolicy.Tls)
+				}
+			}
+		}
+	}
+	if vnode.Spec.BackendDefaults != nil {
+		if vnode.Spec.BackendDefaults.ClientPolicy != nil {
+			clientPolicy := vnode.Spec.BackendDefaults.ClientPolicy
+			if clientPolicy.Tls != nil {
+				mergeTlsClientPolicyDefaults(clientPolicy.Tls)
+			}
+		}
+	}
+}
+
+func mergeTlsClientPolicyDefaults(baseClientPolicyTls *appmeshv1beta1.ClientPolicyTls) {
+	if baseClientPolicyTls == nil {
```

Comment:
redundant null check

Reply:
I assume you're suggesting removing the null check at the call site, not here.

Reply:
i.e. the call site becomes
```
if vnode.Spec.BackendDefaults != nil {
	mergeTlsClientPolicyDefaults(vnode.Spec.BackendDefaults.ClientPolicy)
}
```

Or would it be more appropriate to pass in a value and return a new copy?

Reply:
i suggest remove it from function side. it's more like io.read(buff) where buff is expected to be a valid buffer.